Add Mux Video

mmcc opened the pull request:
👋 Hi there!

Totally open to feedback on this one, so please let me know if there's stuff that needs to change. Particularly the ReadMe was intended to be used on the "[marketing site](https://contentful.mux.dev)" for this with it being deployed to our servers, so let me know if there's a pattern I should be following there.

Also, I've noticed that occasionally when the asset state switches to ready and the preview player shows, the iframe doesn't resize so you end up with a small, scrolling window with the player in it. Should I be doing something to ensure that the iframe resizes when the content grows?

suevalov commented on marketplace/mux-video/extension.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "id": "mux-video-uploader",
```
Can you rename extension folder to match the extension ID?

suevalov commented on marketplace/mux-video/README.md:
````diff
@@ -0,0 +1,86 @@
+In the modal, give the URL for this repo's `extension.json` file:
+
+```
+https://github.com/muxinc/examples/blob/master/contentful-uploader/extension.json
````
I believe that this link is misleading. Can you update it?

suevalov commented on marketplace/mux-video/package.json:
```diff
@@ -0,0 +1,61 @@
+  "version": "1.0.0",
+  "private": true,
+  "devDependencies": {
+    "@babel/core": "^7.4.0",
```
By default we use `7.3.4` of this dependency with current `.babelrc` setup. With `7.4.0` I see the following warning during `npm run start`:

```bash
WARNING: We noticed you're using the `useBuiltIns` option without declaring a core-js version. Currently, we assume version 2.x when no version is passed. Since this default version will likely change in future versions of Babel, we recommend explicitly setting the core-js version you are using via the `corejs` option.

You should also be sure that the version you pass to the `corejs` option matches the version specified in your `package.json`'s `dependencies` section. If it doesn't, you need to run one of the following commands:

  npm install --save core-js@2    npm install --save core-js@3
  yarn add core-js@2              yarn add core-js@3
```

Would be nice to address the issue

mmcc replied:
I actually updated in order to try and _fix_ this issue (it did not). I downgraded but still see the warning, so let me know how you want to proceed.

suevalov replied:
I think all of the babel dependencies should match `7.3.*` to remove the warning.

mmcc commented on marketplace/mux-video/README.md:
````diff
@@ -0,0 +1,86 @@
+To install the UI Extension:
+
+```
+npm run install-extension
````
Ah right, I cleaned up some of the scripts from our `examples` repo when I realized this was going to be in the marketplace with click-to-install rather than needing to be installed via the CLI.

Before I go in and just remove those lines from the Readme, what do you think? I assume it's superfluous given it'd be a rare installation path, but just in case... Here's what `install-extension` currently does in our examples repo:

```
"install-extension": "eval `cat .env` && contentful extension create --installation-parameters '{\"muxAccessTokenId\": \"'$MUX_TOKEN_ID'\", \"muxAccessTokenSecret\": \"'$MUX_TOKEN_SECRET'\"}'"
```

mmcc commented:
Thanks for the review!

[Here's a video](http://take.ms/Eu8Up) that both shows the plugin in action and shows the issue I'm seeing with the resizer not working occasionally. I'm going to hold off on the README changes per the questions, namely what should we be telling people about installing since this is in the marketplace?

suevalov commented:
Thanks for the video! I'd suggest to use the following function https://github.com/contentful/ui-extensions-sdk/blob/master/typings.d.ts#L240 to update the height of the extension once video is loaded.

Yep, let's update README when we merge it and put it on Marketplace.

mmcc commented:
Ok so took another few passes at this, but I think it's the computed `video` element height that are causing us issues and `updateHeight` nor the `autoResizer` don't seem to be helping. I started to go down the path of manually setting a height on the element to its computed height, but I'm starting to think that might be more trouble than it's worth. The `min-height` in CSS at least ensures the content is usable on the initial transition, and any subsequent loads have the correct dimensions. What do you think?

mmcc commented:
Ok so it's a little hacky, but it seems to fix it. On render I have the player set a fixed height for itself based on the aspect ratio that comes back from Mux and the computed width of the video element at 100%. I'm still firing `updateHeight` in the player's `onReady` callback, but it _should_ be superfluous now.

So the only other changes now are the README, and it sounds like we're going to wait and just update those with instructions on installing from the marketplace?

mmcc commented:
Hey @suevalov, just checking in on this one. Let me know if you have any thoughts on the updates!

suevalov commented:
@mmcc Sorry for the late response, I was out of the office almost the whole week.

Looks good to me. Now it's on our side to add it to https://www.contentful.com/developers/marketplace/

mmcc commented:
@suevalov no worries at all. Don't hesitate to ping me if there's anything else I can do to help.

suevalov commented:
Hey @mmcc! Sorry for a long delay but we're ready to place the extension on the Marketplace. Could you please rebase you branch and resolve conflicts?

realityking commented on package.json:
```diff
@@ -12,9 +12,10 @@
     "eslint-plugin-react": "^7.12.4",
-    "lint-staged": "8.1.5",
     "husky": "1.3.1",
+    "jest": "^24.7.1",
```
Why is jest added to the global package.json?

realityking commented on marketplace/mux-video-uploader/extension.json:
```diff
@@ -0,0 +1,24 @@
+  "id": "mux-video-uploader",
+  "name": "Mux Video Uploader",
+  "src": "mux-video-uploader-1.contentfulexts.com",
+  "fieldTypes": ["Object"],
```
Why is this using the field type `Object`? As far as I could tell is stores a simple ID which could be a `Symbol`.

mmcc replied:
I could be misunderstanding something, but as far as I can tell this is an object. The [`MuxContentfulObject` interface](https://github.com/contentful/extensions/pull/118/files#diff-5f6c0a600833ccc258e7933b210cc871R21) describes the field object.

mmcc commented:
Rebased and fixed what seems to be an errant `Jest` import pointed out by @realityking (thanks!). Let me know if you all want me to squash or anything along those lines, the commit history got a little messy there.
